### PR TITLE
gnome-resources: init at 1.2.1

### PR DESCRIPTION
**Pull request by EwuUwe**
## Description of changes
- Add [Resources](https://github.com/nokyan/resources). A modern system monitor designed for the GNOME desktop.
- Add myself as a maintainer

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/gnome-resources/default.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  src = fetchFromGitHub {
+    owner = "nokyan";
+    repo = "resources";
+    rev = "v${version}";
+    hash = "sha256-faZ6MDOu/y4+DX9ObjPyVkxKwmLffMJZ93Adf/t8nxQ=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = "${src}/Cargo.lock";
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:33]: This is IFD which is not allowed. You need to put the Cargo.lock in nixpkgs.

File: pkgs/tools/system/gnome-resources/default.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  ];
+
+  wrapperPath = lib.makeBinPath ([
+    dmidecode
+  ]);
+
+  postFixup = ''
+    # Ensure all dependencies are in PATH
+    wrapProgram $out/bin/resources \
+      --prefix PATH : "${wrapperPath}"
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:66]: This introduces double wrapper. At least you should add it to the gapps wrapper. I would prefer to patch the path directly. See https://github.com/NixOS/nixpkgs/pull/272899/files#diff-fae1573573a7335447ad23a3de60070fe61024bc57067ccb4a69e64ad1d5d1eeR112 for example.

File: pkgs/tools/system/gnome-resources/default.nix [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/nokyan/resources";
+    description = "Monitor your system resources and processes";
+    license = licenses.gpl3;
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:72]:
```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/tools/system/gnome-resources/default.nix [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+, systemd
+, polkit
+, dmidecode
+, gtk4
+, libadwaita
+, ninja
+}:
+stdenv.mkDerivation rec {
+  pname = "gnome-resources";
+  version = "1.2.0";
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:23]:
```suggestion
  version = "1.2.1";
```

File: pkgs/tools/system/gnome-resources/default.nix [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+  src = fetchFromGitHub {
+    owner = "nokyan";
+    repo = "resources";
+    rev = "v${version}";
+    hash = "sha256-faZ6MDOu/y4+DX9ObjPyVkxKwmLffMJZ93Adf/t8nxQ=";
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:29]:
```suggestion
    hash = "sha256-OVz1vsmOtH/5sEuyl2BfDqG2/9D1HGtHA0FtPntKQT0=";
```

File: pkgs/tools/system/gnome-resources/default.nix [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = "${src}/Cargo.lock";
+    outputHashes = {
+      "plotters-cairo-0.4.0" = "sha256-m7ZT5F7WxjZSGQwuytqdMWUgYvcK2UCU/ntJGIJE+UA=";
+    };
+  };
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:37]:
```suggestion
  cargoDeps = rustPlatform.fetchCargoTarball {
    inherit src;
    hash = "sha256-SkCEA9CKqzy0wSIUj0DG6asIysD7G9i3nJ9jwhwAUqY=";
  };
```

File: pkgs/tools/system/gnome-resources/default.nix [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  wrapperPath = lib.makeBinPath ([
+    dmidecode
+  ]);
+
+  postFixup = ''
+    # Ensure all dependencies are in PATH
+    wrapProgram $out/bin/resources \
+      --prefix PATH : "${wrapperPath}"
+  '';
+
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:68]:
```suggestion
```

File: pkgs/tools/system/gnome-resources/default.nix [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  ];
+
+  buildInputs = [
+    glib
+    gtk4
+    libadwaita
+    polkit
+    systemd
+  ];
+
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:58]:
```suggestion

  mesonFlags = [
    "-Dprofile=default"
  ];

```

File: pkgs/tools/system/gnome-resources/default.nix [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  pname = "gnome-resources";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "nokyan";
+    repo = "resources";
+    rev = "v${version}";
+    hash = "sha256-faZ6MDOu/y4+DX9ObjPyVkxKwmLffMJZ93Adf/t8nxQ=";
+  };
+
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:31]:
```suggestion

  postPatch = ''
    substituteInPlace src/utils/memory.rs \
      --replace '"dmidecode"' '"${dmidecode}/bin/dmidecode"'
  '';

```

File: pkgs/tools/system/gnome-resources/Cargo.lock
Comment (jtojnar) [inline pkgs/tools/system/gnome-resources/Cargo.lock:1]: This does not appear to be used.

**Comment by linsui** [comment]
Please remove Cargo.lock. It's not needed since there is no git deps and use fetchCargoTarball instead.

File: pkgs/tools/system/gnome-resources/default.nix [hunk 10]
```diff
@@ -0,0 +1,72 @@
+{ pkgs
+, fetchFromGitHub
```
Comment (linsui) [inline pkgs/tools/system/gnome-resources/default.nix:2]:
```suggestion
{ fetchFromGitHub
```

**Review by linsui — APPROVED** [review]

**Comment by jtojnar** [comment]
For the next time, the commits should be squashed according to contributing guide before merging.

**Comment by Lukas-Heiligenbrunner** [comment]
Hey, I hope you know this is a duplicate of https://github.com/NixOS/nixpkgs/pull/270250

**Comment by linsui** [comment]
Hmmm, right. I thought we should pick the dmidecode patch there and add @EwuUwe as a maintainer, then remove this one.

**Comment by Lukas-Heiligenbrunner** [comment]
Is this patch really necessary? Its seems to be working fine without...?

**Comment by linsui** [comment]
Check the memory panel. There is a Authenticate banner but the it doesn't works without the patch.

**Comment by amaxine** [comment]
Sorry, I missed the commit mess. I think the right fix here is to pick that patch but drop the name `gnome-resources` (with an alias, since I did introduce it for a short while), as the upstream name seems to only be `resources` even if it's primarily a gnome app.

**Comment by Lukas-Heiligenbrunner** [comment]
> Check the memory panel. There is a Authenticate banner but the it doesn't works without the patch.

Yeah, I know. For me its working fine without the patch. 
In my case it detects dmidecode at `/run/current-system/sw/bin/dmidecode`
(But might be a special thing on my side idk)

**Comment by linsui** [comment]
It should work even if you don't have dmidecode in your path.

**Pull request merged**